Tests/mock_server.py

```diff
@@ -289,7 +294,8 @@ def start(self, playbook_id, path=None, record=False):
         if not log_file_exists:
             self.stop()
             raise Exception("Proxy process took to long to go up.")
-        print('Proxy process up and running. Took {} seconds'.format(seconds_since_init))
+        proxy_up_message = 'Proxy process up and running. Took {} seconds'.format(seconds_since_init)
+        prints_manager.add_print_job(proxy_up_message, print, thread_index)
```

Comment:
In several functions headers, `prints_manager=None` is being used, however, you are never addressing this case where `prints_manage` is actually `None`. For example here, you have `prints_manager.add_print_job(proxy_up_message, print, thread_index)` which could obviously fail if `prints_manager` is `None. It feels to me that we either should address this case where `prints_manager=None` or don't have the default at all. Tell me what you think.
This comment is relevant for every function that has `prints_manager=None` as a parameter.